clients/lsp-clangd.el

```diff
@@ -178,6 +178,49 @@ This must be set only once after loading the clang client.")
   :risky t
   :type '(repeat string))
 
+(defun lsp-clangd-find-compile-commands-dir ()
+  "Return the full directory path (or nil) that contain a compile_commands.json.
+
+If multiple compile_commands.json files are
+found in the project directory tree - ask the user to choose."
+  ;; TODO lsp-workspace-root won't find anything at the start of the workspace
+  (let ((candidate-dirs (directory-files-recursively (lsp-workspace-root) "\\`compile_commands.json\\'")))
```

Comment:
This is expensive looking up, especially in large project and HDD, which can potentially freeze Emacs.
Can we change the heuristic here?
- Looking at the root workspace only
- Or looking using `locate-dominating-file` from current open file?

Alternatively, can we run these command in background by calling it from shell (`start-file-process-shell-command`) and parse the returns or using `[async.el](https://github.com/jwiegley/emacs-async)`?
So that user will not be blocked when open a `.cpp` file

Reply:
> Looking at the root workspace only

clangd looks up compile_commands.json in the project root by default. This auto-configure is precisely to improve on this naive heuristic for out of source cmake builds, where compile_commands.json will be in another directory. 

You and yyoncho have raised very valid points and thanks for the suggestions - I will try to do my best with minimising the runtime cost.